sdk/v2/core/jobs.go

```diff
@@ -0,0 +1,381 @@
+package core
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/brigadecore/brigade/sdk/v2/internal/restmachinery"
+	"github.com/brigadecore/brigade/sdk/v2/meta"
+)
+
+// JobPhase represents where a Job is within its lifecycle.
+type JobPhase string
+
+const (
+	// JobPhaseAborted represents the state wherein a Job was forcefully
+	// stopped during execution.
+	JobPhaseAborted JobPhase = "ABORTED"
+	// JobPhaseFailed represents the state wherein a Job has run to
+	// completion but experienced errors.
+	JobPhaseFailed JobPhase = "FAILED"
+	// JobPhasePending represents the state wherein a Job is awaiting
+	// execution.
+	JobPhasePending JobPhase = "PENDING"
+	// JobPhaseRunning represents the state wherein a Job is currently
+	// being executed.
+	JobPhaseRunning JobPhase = "RUNNING"
+	// JobPhaseSucceeded represents the state where a Job has run to
+	// completion without error.
+	JobPhaseSucceeded JobPhase = "SUCCEEDED"
+	// JobPhaseTimedOut represents the state wherein a Job has has not completed
+	// within a designated timeframe.
+	JobPhaseTimedOut JobPhase = "TIMED_OUT"
+	// JobPhaseUnknown represents the state wherein a Job's state is unknown. Note
+	// that this is possible if and only if the underlying Job execution substrate
+	// (Kubernetes), for some unanticipated, reason does not know the Job's
+	// (Pod's) state.
+	JobPhaseUnknown WorkerPhase = "UNKNOWN"
+)
+
+// Job represents a component spawned by a Worker to complete a single task
```

Comment:
@vdice Good question. And I think this raises a slight design change that might warrant further discussion.

cc @radu-matei 

If you look at how jobs are now defined here (`JobSpec`s, actually), they no longer take an array of "tasks" as they once did. Instead they have a `Command` array and an `Arguments` array.

The primary reason I changed this is that the 1.x worker did a fair amount of magic to take an array of discrete "tasks" and turn it into a shell script-- and in doing so made fairly large assumptions that Linux containers all include `bash` and that Windows containers all include PowerShell. These assumptions aren't universally valid and they specifically preclude the possibility of using a lot of small, simple images that _don't_ include those shells. 2.0 workers will use the API to push Job creation responsibility to the API server, and rather than port those large assumptions to the backend, I made the call to change up how jobs are defined.

The `Command` array and an `Arguments` array I previously referred to _admittedly_ may end up feeling clumsier to work with, _but_ it's in 100% parity with how Docker CLI, Docker packages, Docker Compose configuration files, and Kubernetes pods ask one to define what is to be executed within the container. i.e., the new approach reuses a familiar and widespread precedent.

So... in the end, I think it's not misleading to say a Job represents a "single task."

With proper syntax, one can of course effect execution of multiple commands and there is always the option to use a single command to invoke a script-- which, not insignificantly, is made all the more easier by the fact that arbitrary files are so easily embedded into a project definition now-- a useful pattern that should absolutely be covered in end-user docs.

Reply:
Ah, thank you for the explanation.  I'm persuaded that moving to `Command` and `Arguments` is the right choice, thanks to the reasoning above.  Wording indeed seems okay to stay as is.